src/from_sql.rs

```diff
@@ -0,0 +1,96 @@
+use crate::{tds::Numeric, xml::XmlData, ColumnData};
+use uuid::Uuid;
+
+/// A conversion trait from a TDS type by-reference.
+pub trait FromSql<'a>
+where
+    Self: Sized + 'a,
+{
+    /// Returns the value, `None` being a null value, copying the value.
+    fn from_sql(value: &'a ColumnData<'static>) -> crate::Result<Option<Self>>;
```

Comment:
I wonder why the `'static` here, since there is a lifetime at hand (I haven't looked up what the ColumnData lifetime param refers to, may be irrelevant).

Reply:
I looked at ColumnData, and it makes sense to have `'static` here if we always get owned strings from the wire.

Reply:
Yep. All the data comes in little-endian format anyhow and strings in UCS2, so we always take the ownership, hence the `'static` lifetime.